crates/modelardb_manager/src/remote.rs

```diff
@@ -150,24 +154,108 @@ impl FlightService for FlightServiceHandler {
         Err(Status::unimplemented("Not implemented."))
     }
 
-    /// TODO: Perform a specific action based on the type of the action in `request`.
-    ///       The following actions should be supported:
-    ///       * `CommandStatementUpdate`
-    ///       * `UpdateRemoteObjectStore`
-    ///       * `RegisterEdge`
-    ///       * `RemoveEdge`
+    /// Perform a specific action based on the type of the action in `request`. Currently the
+    /// following actions are supported:
+    /// * `RegisterNode`: Register either an edge or cloud node with the manager. The node is added
+    /// to the cluster of nodes controlled by the manager and the object store and current database
+    /// schema used in the cluster is returned.
+    /// * `RemoveNode`: Remove a node from the cluster of nodes controlled by the manager and
+    /// kill the process running the node. The specific node to remove is given through the
+    /// uniquely identifying URL of the node.
     async fn do_action(
         &self,
-        _request: Request<Action>,
+        request: Request<Action>,
     ) -> Result<Response<Self::DoActionStream>, Status> {
-        Err(Status::unimplemented("Not implemented."))
+        let action = request.into_inner();
+        info!("Received request to perform action '{}'.", action.r#type);
+
+        if action.r#type == "RegisterNode" {
```

Comment:
It is unclear how the system recovers if `modelardb_manager` encounters an error half-way through registering or removing a node, maybe add a comment that documents this as it currently looks like the cluster can be left in a problematic state if the `ClusterManager` and `MetadataManager` disagrees about which nodes exist or they think one has been removed but the kill command was not send to it.

Reply:
Good point, I think it is a hard problem to fix completely but I have added some comments to make it clear that it is a possibility.

Reply:
Ok, I have added a task to continue working on this problem later.